2024/aoc/__main__.py

```diff
@@ -0,0 +1,34 @@
+import datetime
+
+import click
+
+from . import days
+
+
+@click.command()
+@click.option(
+    "-i",
+    "--input",
+    "data",
+    type=click.File(mode="rt", encoding="utf8"),
+    default="-",
+    help="Problem input file",
+)
+@click.option("-t", "--time", is_flag=True, help="Print elapsed time afterwards")
+@click.argument("day", required=True)
+def main(day: int, time: bool, data: str) -> None:
+    runner_class = days.get_runner(day)
+
+    start = datetime.datetime.now()
```

Comment:
You likely want [`perf_counter`](https://docs.python.org/3/library/time.html#time.perf_counter). But even more likely you don't want to do the timings in Python :)

Reply:
I do want `perf_counter`! Or rather the nano version, for accuracy But I also think I want to do the timing inside python, as it allows me to discount overhead that I don't particularly care for (interpreter startup, loading data, loading modules)